app/models/team.rb

```diff
@@ -4,14 +4,21 @@ class Team < ActiveRecord::Base
 
   has_many :namespaces
 
-  # Users & owners
   has_many :team_users
   has_many :users, through: :team_users
-  has_many :owners, -> { where 'team_users.owner': true },
+  has_many :owners, -> { where 'team_users.role' => TeamUser.roles['owner'] },
+    through: :team_users, source: :user
+  has_many :contributors, -> { where 'team_users.role' => TeamUser.roles['contributor'] },
+    through: :team_users, source: :user
+  has_many :viewers, -> { where 'team_users.role' => TeamUser.roles['viewer'] },
```

Comment:
I believe that `viewers` won't be used. That's because any member of the team has effectively pull access, so the viewers relationship is redundant here (see the `NamespacePolicy::pull?` method from your PR).